src/m365/aad/commands/group/group-user-list.ts

```diff
@@ -0,0 +1,152 @@
+    return ['id', 'displayName', 'userPrincipalName', 'userType'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initOptionSets();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupDisplayName: typeof args.options.groupDisplayName !== 'undefined',
+        role: typeof args.options.role !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: "-i, --groupId [groupId]"
+      },
+      {
+        option: "-n, --groupDisplayName [groupDisplayName]"
+      },
+      {
+        option: "-r, --role [type]",
+        autocomplete: ["Owner", "Member", "Guest"]
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      {
+        options: ['groupId', 'groupDisplayName']
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.role) {
+          if (['Owner', 'Member', 'Guest'].indexOf(args.options.role) === -1) {
+            return `${args.options.role} is not a valid role value. Allowed values Owner|Member|Guest`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const groupId = await this.getGroupId(args.options);
+      let users = await this.getOwners(groupId, logger);
+
+      if (args.options.role !== 'Owner') {
+        const membersAndGuests = await this.getMembersAndGuests(groupId, logger);
+        users = users.concat(membersAndGuests);
+      }
+
+      if (args.options.role) {
+        users = users.filter(i => i.userType === args.options.role);
+      }
+
+      await logger.log(users);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getGroupId(options: Options): Promise<string> {
+    const { groupId, groupDisplayName } = options;
+
+    if (groupId) {
+      return groupId;
+    }
+
+    return await aadGroup.getGroupIdByDisplayName(groupDisplayName!);
+  }
+
+  private async getOwners(groupId: string, logger: Logger): Promise<User[]> {
+    if (this.verbose) {
+      await logger.logToStderr(`Retrieving owners of the group with id ${groupId}`);
+    }
+
+    const endpoint: string = `${this.resource}/v1.0/groups/${groupId}/owners?$select=id,displayName,userPrincipalName,userType`;
+
+    const users = await odata.getAllItems<User>(endpoint);
+
+    // Currently there is a bug in the Microsoft Graph that returns Owners as
```

Comment:
Hi @nanddeepn, this is not really a bug. The `userType` property does not reflect the relationship of the user to this AAD group, but the relationship to the directory. 

https://learn.microsoft.com/en-us/azure/active-directory/hybrid/cloud-sync/how-to-map-usertype

Hence you are here mixing different purposes. We should definitely not overwrite the `userType` property. I believe it's fine to include it in the output, as it's useful to know how many guests are in a group. But it may be a bit confusing as well, as people may mix it with the values in the role option. 

I need to think about this one...

Reply:
I'm afraid Martin is right here. I can take the blame here because I kind of copied options from `aad m365group user list` without paying too much attention to the code, sorry! What we do over there is definitely not a good idea.
What we could do is introduce a new property (`role`?) to show which user is Owner and which one is Member. But we should definitely leave the `userType` untouched as this is a property from the user object.
I'm afraid this will be a breaking change to get this fixed for `aad m365group user list` @martinlingstuyl 🫤. Good catch to notice this.